.github/workflows/mobile-compile_time_options.yml

```diff
@@ -27,17 +27,43 @@ jobs:
     - uses: actions/checkout@v3
     - name: Add safe directory
       run: git config --global --add safe.directory /__w/envoy/envoy
-    - name: 'Building C++ library'
+    - name: 'Running C++ test with YAML disabled'
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       # Envoy Mobile build which verifies that the build configuration where YAML is disabled.
       run: |
-        cd mobile && ./bazelw test \
+        cd mobile
+        ./bazelw test \
             $([ -z $GITHUB_TOKEN ] || echo "--config=remote-ci-linux") \
             --config=ci \
             --define=envoy_yaml=disabled \
             --test_env=ENVOY_IP_TEST_VERSIONS=v4only \
             //test/common/integration:client_integration_test --test_output=all
+  cc_test:
+    needs: env
+    name: cc_test
+    runs-on: ubuntu-20.04
+    timeout-minutes: 120
+    container:
+      image: envoyproxy/envoy-build-ubuntu:41c5a05d708972d703661b702a63ef5060125c33
+    steps:
+    - uses: actions/checkout@v3
+    - name: Add safe directory
+      run: git config --global --add safe.directory /__w/envoy/envoy
+    - name: 'Running C++ tests'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        cd mobile
+        ./bazelw test \
+            $([ -z $GITHUB_TOKEN ] || echo "--config=remote-ci-linux") \
+            --config=ci \
+            --define=signal_trace=disabled \
+            --define=envoy_mobile_request_compression=disabled \
+            --define=envoy_enable_http_datagrams=disabled \
+            --define=google_grpc=disabled \
+            --@com_envoyproxy_protoc_gen_validate//bazel:template-flavor= \
+            $(bazel query //test/cc/... + //test/common/... except //test/common/integration:client_integration_test)
```

Comment:
seems this is pulling more tests in than before and (at least) one is failing

```console

//test/common/integration:sds_integration_test                           FAILED in 2.8s
  /github/home/.cache/bazel/_bazel_root/4fefd9a7510a7344d47b58c4ccb5a56e/execroot/envoy_mobile/bazel-out/k8-fastbuild/testlogs
```

with the way mobile ci runs its not immediately obvious what the failure is - also not sure whether the test/run requires some adjustment or whether it just shouldnt run this test

Reply:
I purposely wanted to add more tests in to catch issues in the test code with different (compile time) configurations.

It looks like the intended goal worked b/c this caught a problem in that sds_integration_test wasn't surrounded by a `envoy_select_google_grpc`!

so I added that to the sds_integration_test target and now it's passing.

PTAL!